src/algebra/quaternion.lean

```diff
@@ -84,12 +84,21 @@ def equiv_tuple {R : Type*} (c₁ c₂ : R) : ℍ[R, c₁, c₂] ≃ (fin 4 →
 
 variables {S T R : Type*} [comm_ring R] {c₁ c₂ : R} (r x y z : R) (a b c : ℍ[R, c₁, c₂])
 
+/-- The imaginary part of a quaternion. -/
+def im (x : ℍ[R, c₁, c₂]) : ℍ[R, c₁, c₂] := ⟨0, x.im_i, x.im_j, x.im_k⟩
+
+@[simp] lemma im_re : a.im.re = 0 := rfl
+@[simp] lemma im_im_i : a.im.im_i = a.im_i := rfl
+@[simp] lemma im_im_j : a.im.im_j = a.im_j := rfl
+@[simp] lemma im_im_k : a.im.im_k = a.im_k := rfl
+@[simp] lemma im_idem : a.im.im = a.im := rfl
+
 instance : has_coe_t R (ℍ[R, c₁, c₂]) := ⟨λ x, ⟨x, 0, 0, 0⟩⟩
 
-@[simp] lemma coe_re : (x : ℍ[R, c₁, c₂]).re = x := rfl
-@[simp] lemma coe_im_i : (x : ℍ[R, c₁, c₂]).im_i = 0 := rfl
-@[simp] lemma coe_im_j : (x : ℍ[R, c₁, c₂]).im_j = 0 := rfl
-@[simp] lemma coe_im_k : (x : ℍ[R, c₁, c₂]).im_k = 0 := rfl
+@[simp, norm_cast] lemma coe_re : (x : ℍ[R, c₁, c₂]).re = x := rfl
+@[simp, norm_cast] lemma coe_im_i : (x : ℍ[R, c₁, c₂]).im_i = 0 := rfl
+@[simp, norm_cast] lemma coe_im_j : (x : ℍ[R, c₁, c₂]).im_j = 0 := rfl
+@[simp, norm_cast] lemma coe_im_k : (x : ℍ[R, c₁, c₂]).im_k = 0 := rfl
 
```

Comment:
Any reason why we can't have `coe_im` here rather than on l. 143?

Reply:
Yes, we have no `has_zero` yet for the RHS.